experimental/Schemes/SpecOpen.jl

```diff
@@ -653,7 +653,22 @@ the numerator and denominator of ``f`` have to be elements of
 the ring ``𝕜[x₁,…,xₙ]``.
 """
 function maximal_extension(
-    X::AbsSpec{<:Ring, <:AbsLocalizedRing}, 
+    X::AbsSpec{<:Ring, <:MPolyLocalizedRing}, 
+    f::AbstractAlgebra.Generic.Frac{RET}
+  ) where {RET<:RingElem}
+
+  a = numerator(f)
+  b = denominator(f)
+  W = OO(X)
+  I = quotient(ideal(W, b), ideal(W, a))
```

Comment:
Can you explain why this is the way to go? Can't we just do a gcd in the ambient polynomial rings?

Reply:
No, unfortunately not. The idea is to produce an ideal that describes the full locus on which `f` is necessarily ill-defined. This follows some parts of Janko's [script](http://www.mathematik.uni-kl.de/~boehm/lehre/17_CA/ca.pdf). In particular, this `quotient` makes a difference when `W` itself is a quotient ring or a localization thereof.

Reply:
Exactly. But here `X::AbsSpec{<:Ring, <:MPolyLocalizedRing}` and so `W` should be `MPolyLocalizedRing` 
hence there is no quotient and we are working in some localization of a unique factorization domain and there everything should be easy?

Reply:
Oh. I didn't pay attention to that, sorry. Yes, then what you say probably makes sense. Can you change that? Or should I do it? We might even consider returning a function on a `PrincipalOpenSubset` in this case.